src/monitor.ts

```diff
@@ -53,6 +53,8 @@ export const setGlobalOptions = ({ logExecutionStart, logResult, parseError, pro
   global.prometheusBuckets = prometheusBuckets;
 };
 
+const isAPromiseLike = <T>(obj: any): obj is PromiseLike<T> => typeof obj === 'object' && typeof obj.then === 'function';
```

Comment:
let's use https://github.com/sindresorhus/is it has a `is.promise()`
also,
https://blog.sindresorhus.com/small-focused-modules-9238d977a92a